apps/ledger-live-desktop/src/renderer/components/DeviceAction/index.tsx

```diff
@@ -364,7 +374,67 @@ export const DeviceActionDefaultRendering = <R, H extends States, P>({
     }
   }, [dispatch, device, deviceInfo, latestFirmware]);
 
-  if (displayUpgradeWarning && appAndVersion && passWarning) {
+  if (deprecate && deprecateData && onContinue) {
+    const { deprecatedFlowExceptions, tokenExceptions, coin, date, modelId } = deprecateData;
+    const currentFlow = (() => {
+      switch (location) {
+        case HOOKS_TRACKING_LOCATIONS.genericDAppTransactionSend:
+          return "stake";
+        case HOOKS_TRACKING_LOCATIONS.sendModal:
+          return "send";
+        case HOOKS_TRACKING_LOCATIONS.receiveModal:
+          return "receive";
+        case HOOKS_TRACKING_LOCATIONS.exchange:
+          return "swap";
+        default:
+          return "other";
+      }
+    })();
+    const skippedDeprecations = stateSettings.deprecationDoNotRemind;
+    const alreadyDismissed = skippedDeprecations.includes(coin);
+    const currentToken =
+      request && typeof request === "object" && "tokenCurrency" in request && request.tokenCurrency
+        ? (request.tokenCurrency as TokenCurrency).name
+        : "undefined";
```

Comment:
Pretty uncommon to hardcode the string`"string"`. How about simply
```suggestion
        : "";
```